tests/cases/fourslash/codeFixRequireInTs2.ts

```diff
@@ -0,0 +1,6 @@
+/// <reference path='fourslash.ts' />
+
+// @Filename: /a.ts
+////const { a } = [|require("a")|];
+
+verify.not.codeFixAvailable();

```

Comment:
I feel like maybe I’m missing something or my morning coffee hasn’t kicked in yet, but at first glance I don’t understand the connection between `allowSyntheticDefaultImports` and named bindings making the codefix not available. Since we don’t know anything about module `"a"`, it seems like it could be shaped like

```ts
export const a: string;
```

Which from JS we’d let you use `const { a } = require("a")` and in TS we’d let you use `import { a } from "a"` regardless of allowSynthetic/esModuleInterop, right?

Reply:
I think your coffee is good 👍.

Reply:
So why isn’t this one available as a codefix now? Shouldn’t we get `import { a } from "a"` on this?

Reply:
> Actually, I think the right behaviour would be to not offer the codefix if any element is not an identifier.

I removed logic which tried to convert `ObjectBindingPattern` to named `NamedImports`.